### ISSUE TITLE
Running in PyCharm throws termios.error: (5, 'Input/output error')

### ISSUE THREAD

**Issue by sla-te** [IngoMeyer441/simple-term-menu#81]
Start Pycharm, create a new file with the following content and hit "Run"

```python
def main():
    terminal_menu = simple_term_menu.TerminalMenu(["entry 1", "entry 2", "entry 3"])
    menu_entry_index = terminal_menu.show()
    print(f"You choose: {menu_entry_index}")

if __name__ == "__main__":
    main()
```

```
Traceback (most recent call last):
  File "/.../tmp.py", line 19, in <module>
    main()
  File "/.../tmp.py", line 13, in main
    menu_entry_index = terminal_menu.show()
                       ^^^^^^^^^^^^^^^^^^^^
  File ".../venv/lib/python3.11/site-packages/simple_term_menu.py", line 1497, in show
    self._init_term()
  File ".../venv/lib/python3.11/site-packages/simple_term_menu.py", line 961, in _init_term
    termios.tcsetattr(
termios.error: (5, 'Input/output error')
****

**Comment by IngoMeyer441** [comment]
Thanks for the bug report. By default, PyCharm does not create a full terminal instance which is needed by simple-term-menu, thus raising the error, you got. You can enable emulating a full terminal emulator in the output console as a workaround. See <https://intellij-support.jetbrains.com/hc/en-us/community/posts/360003383619/comments/6589796593042> for details.

**Comment by IngoMeyer441** [comment]
Ok I see. But the main problem is, that simple-tem-menu relies on being run in a terminal emulator. Without the terminal emulation option, pyCharm uses simple stdin and stdout redirection (as far as I know) and simple-term-menu cannot work with that.

**Comment by IngoMeyer441** [comment]
How shall the debug mode look like?

**Comment by IngoMeyer441** [comment]
Hmm, difficult, since simple-term-menu needs terminal capabilities for redrawing the screen... maybe it can be combined with [pyte](https://pyte.readthedocs.io/en/latest/index.html) in this special case? I am quite unsure about this.

**Comment by IngoMeyer441** [comment]
Ah ok, I thought the IDE would only crash sometimes with terminal emulation mode turned on. So it crashes without terminal emulation?

**Comment by IngoMeyer441** [comment]
I have run some tests in PyCharm. In the latest PyCharm version, the output console seems to always have some kind of terminal emulation and terminal emulation cannot be switched on separately. However, no menu is shown :disappointed:. Therefore, I have added a check for the PyCharm output console on branch [`fix/pycharm-console`](https://github.com/IngoMeyer441/simple-term-menu/tree/fix/pycharm-console) and abort the execution. Is this ok for now? You can test this change by getting the patched package version:

```bash
pip install 'git+https://github.com/IngoMeyer441/simple-term-menu@fix/pycharm-console'
```

**Comment by IngoMeyer441** [comment]
The `emulate terminal output in console` checkbox is missing in the latest PyCharm release. Do you still use an older version?

**Comment by IngoMeyer441** [comment]
Thanks, I didn't find this.
I have updated the branch and added a check for the `TERM` variable. It can be used to detect a terminal in the PyCharm output console.
However, there is still an issue left: If PyCharm itself is started from a terminal emulator, the `TERM` variable is always set and the detection won't work.

**Comment by IngoMeyer441** [comment]
The fix is added in release [v1.6.2](https://github.com/IngoMeyer441/simple-term-menu/releases/tag/v1.6.2). If it still needs improvement, please feel free to reopen.
